pcap4j-core/src/main/java/org/pcap4j/packet/IcmpV6HomeAgentAddressDiscoveryReplyPacket.java

```diff
@@ -0,0 +1,266 @@
+                sb.append("The data is too short to build an ICMPv6 Home Agent Address Discovery Reply Header(")
+                        .append(ICMPV6_HOME_AGENT_ADDRESS_DISCOVERY_REPLY_HEADER_MIN_SIZE)
+                        .append(" bytes). data: ")
+                        .append(ByteArrays.toHexString(rawData, " "))
+                        .append(", offset: ")
+                        .append(offset)
+                        .append(", length: ")
+                        .append(length);
+                throw new IllegalRawDataException(sb.toString());
+            }
+            this.identifier = ByteArrays.getShort(rawData, IDENTIFIER_OFFSET + offset);
+            this.reserved = ByteArrays.getShort(rawData, RESERVED_OFFSET + offset);
+            this.homeAgentAddresses = new ArrayList<Inet6Address>();
+            for (int i = HOME_AGENT_ADDRESSES_OFFSET; i < length; i += INET6_ADDRESS_SIZE_IN_BYTES) {
+                homeAgentAddresses.add(ByteArrays.getInet6Address(rawData, i + offset));
+            }
+        }
+
+        private IcmpV6HomeAgentAddressDiscoveryReplyHeader(Builder builder) {
+            this.identifier = builder.identifier;
+            this.reserved = builder.reserved;
+            this.homeAgentAddresses = builder.homeAgentAddresses;
+        }
+
+        public short getIdentifier() {
+            return identifier;
+        }
+
+        public short getReserved() {
+            return reserved;
+        }
+
+        public List<Inet6Address> getHomeAgentAddresses() {
+            return homeAgentAddresses;
+        }
+
+        @Override
+        protected List<byte[]> getRawFields() {
+            List<byte[]> rawFields = new ArrayList<byte[]>();
+            rawFields.add(ByteArrays.toByteArray(identifier));
+            rawFields.add(ByteArrays.toByteArray(reserved));
+            Iterator<Inet6Address> iter = homeAgentAddresses.iterator();
+            List<byte[]> homeAgentAddressesByteList = new ArrayList<byte[]>();
+            for (int i = HOME_AGENT_ADDRESSES_OFFSET; i < length(); i += INET6_ADDRESS_SIZE_IN_BYTES) {
+                homeAgentAddressesByteList.add(ByteArrays.toByteArray(iter.next()));
+            }
+            rawFields.add(ByteArrays.concatenate(homeAgentAddressesByteList));
+            return rawFields;
+        }
+
+        @Override
+        public int length() {
+            return homeAgentAddresses.size() * INET6_ADDRESS_SIZE_IN_BYTES + HOME_AGENT_ADDRESSES_OFFSET;
+        }
+
+        @Override
+        protected String buildString() {
+            StringBuilder sb = new StringBuilder();
+            String ls = System.getProperty("line.separator");
+
+            sb.append("[ICMPv6 Home Agent Address Discovery Reply Header (")
+                    .append(length())
+                    .append(" bytes)]")
+                    .append(ls);
+            sb.append("  Identifier: ")
+                    .append(identifier)
```

Comment:
Please add a return value of getIdentifierAsInt() instead of identifier for consistency with IcmpIdentifiablePacket.

Reply:
fixed.